src/components/SearchComponent/index.js

```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+
+import "./styles.css";
+import getCity from "../CustomHooks/useCity";
+import StateCityInformation from "../../common/utils/state-city.json";
+
+const SearchComponent = ({ setCurrentCards, teamInformation }) => {
+  const [nameOrDesignation, setNameOrDesignation] = useState("");
+  const [state, setState] = useState("");
+  const [cityName, setCityName] = useState("");
+
+  const city = getCity(state);
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    if (!cityName) {
+      setCurrentCards(
+        teamInformation.filter(
+          (item) =>
+            (item.name &&
+              item.name
+                .toUpperCase()
+                .includes(nameOrDesignation.toUpperCase())) ||
+            (item.followers &&
+              String(item.followers).includes(nameOrDesignation))
+        )
+      );
+    } else {
+      if (cityName && cityName !== "Select") getSpecificCityUsers(cityName);
+      else alert("City Not selected");
+    }
+  };
+
+  const getSpecificCityUsers = async (cityName) => {
+    const response = await fetch(
+      `https://api.github.com/search/users?q=location%3A${cityName}`
+    );
+    const json = await response.json();
+    let users =
+      json &&
+      json.items.map((userInfo) => ({
+        imageSource: userInfo.avatar_url,
+        name: userInfo.login,
+        location: userInfo.location,
+        company: userInfo.company,
+        followers: userInfo.followers,
+        id: userInfo.login,
+      }));
+
+    setCurrentCards(users);
+  };
+
+  const handleStateChange = (e) => {
+    setState(e.target.value);
+    setCityName("");
+  };
+
+  const handleCityChange = (e) => {
+    setCityName(e.target.value);
+  };
+  return (
+    <form onSubmit={handleOnSubmit}>
+      <div className="searchClass">
+        <input
+          type="text"
+          value={nameOrDesignation}
+          onChange={(event) => {
+            setNameOrDesignation(event.target.value);
+            setCityName("");
+          }}
+        />
+        <button className="buttonClass">Search</button>
```

Comment:
can add type="submit" to the button

Reply:
done. i was searching the difference between type="submit" and without type attribute for button in a form, didnt get enough info in net